Force UTF-8 encoding before pushing so Slack

ushu opened the pull request:
Hello,

I had issues when logging exceptions built from http responsse that come in with non-utf8 encoding and raise Encoding::UndefinedConversionError (basically they have invalid characters...).

This patch forces the encoding to UTF-8 in the LinkFormatter class to avoid this issue, whatever the content of the message.

stevenosloan commented:
Cool, I can see some use in this. If you can address those couple code comments and add a test it'll be ready for a merge.

thanks!

ushu commented:
I updated the branch according to your demands and added specs.
Regards !

ushu commented:
Added fix for Ruby 2.0.0 (Travis had found an issue)

+ I found out that the default behaviour in "replace" mode for String#encode was to use the unicode replacement character for invalid unicode sequence and just removing undefined sequences.
I found that pretty cool so I removed the "replace" parameter from the call to stick to it.

stevenosloan commented:
awesome thanks! I'm at a conference till Friday, but will take a full look
and hopefully merge then.

On Tuesday, April 21, 2015, Aurélien Noce <notifications@github.com> wrote:

> Added fix for Ruby 2.0.0 (Travis had found an issue)
>
>    - I found out that the default behaviour in "replace" mode for
>    String#encode was to use the unicode replacement character for invalid
>    unicode sequence and just removing undefined sequences. I found that pretty
>    cool so I removed the "replace" parameter from the call to stick to it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevenosloan/slack-notifier/pull/26#issuecomment-94698072>
> .
>


-- 



Steven Sloan  |  Designer & Developer

stevenosloan commented:
Sorry for the bit of delay there, work piled up a bit more than expect. Thanks for the patch!
